src/Interface/hiopInterfacePrimalDecomp.hpp

```diff
@@ -142,7 +142,7 @@ class hiopInterfacePriDecProblem
                             const hiopVector& x0,
                             const std::string& mem_space);
 
-    ~RecourseApproxEvaluator();
+    virtual ~RecourseApproxEvaluator();
```

Comment:
IMHO, this is an important change to ensure that the right destructor is called when `delete evaluator` is called from `hiopAlgPrimalDecomp.cpp`.